railties/lib/rails/railtie.rb

```diff
@@ -200,21 +200,23 @@ def method_missing(name, *args, &block)
 
     delegate :railtie_name, to: :class
 
-    def initialize
+    def initialize #:nodoc:
       if self.class.abstract_railtie?
         raise "#{self.class.name} is abstract, you cannot instantiate it directly."
       end
     end
 
-    def configure(&block)
+    def configure(&block) #:nodoc:
       instance_eval(&block)
     end
 
+    # It initializes the Railtie specific configuration that are shared by
+    # railties and the application
```

Comment:
This is used to create the `config` object on Railties, an instance of Railtie::Configuration that is used by Railties and Application to store related configuration.